torch/fx/graph_module.py

```diff
@@ -26,6 +27,24 @@ def patched_getline(*args, **kwargs):
     return _orig_getlines(*args, **kwargs)
 linecache.getlines = patched_getline
 
+def deserialize_graphmodule(root : torch.nn.Module, src : str):
+    """
+    Deserialize a GraphModule given the original `root` module and the generated
+    `forward()` source code (`src`). This will exec() the source of the forward
+    onto the root module to create a well-formed Module with code analogous
+    to the original code. Then it symbolically traces through it to get the
+    GraphModule
+    """
+    root = copy.deepcopy(root)
```

Comment:
why do you need deepcopy here? Isn't the original module just created by pickle?

Reply:
It is now, but inevitably somebody is going to find this function and use it for nefarious purposes, so this is trying to be defensive. It would be surprising for this function to both mutate its argument and return an entirely new Module instance

Reply:
I can just change it to a shallow copy

Reply:
if you get rid of `root` then it won't mutate the argument, right? I.e. `symbolic_trace` will just create new subhierarchy every time only from the submodules that are used. And what mutation are we talking about? Is it just the "tensor constants"? that should be address with `root` elimination and I don't think there are any other mutations.

also if you don't want it to be discoverable - put `_` in front. I find it wasteful to put an extra deepcopy in unpickling path as it's absolutely not needed

Reply:
The mutation I'm referring to is line 46, where we're dumping all of the globals onto the passed-in `root` module. For any arbitrary module that's passed in, it's going to (at the very least) wipe out any existing forward method that's on it

Re: adding a leading underscore, I see that as more of a Python cargo cult. It's not enforced and people use our internal APIs all the time.

Reply:
> if you get rid of root

We haven't done that :p